File: clients/go/examples/tables-alter-type-rename.go
package main

import (
	"context"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
	"github.com/datastax/astra-db-go/v2/astra/table"
)

func main() {
	ctx := context.Background()

	// Get a database
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	// Rename fields in a user-defined type
	err := database.AlterType(ctx, "member", table.RenameTypeFields{
		Fields: map[string]string{
			"name": "first_name", "is_active": "is_member",
		}})
	if err != nil {
		log.Fatal(err)
	}
}

File: clients/typescript/examples/tables-alter-type-rename.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get a database
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});

// Drop a user-defined type
(async function () {
  await database.alterType("member", {
    operation: {
      rename: {
        fields: {
          name: "first_name",
          is_active: "is_member",
        },
      },
    },
  });
})();
